src/upnp_connmgr.c

```diff
@@ -275,8 +346,7 @@ static char* str_append_no_termination(char *dest, const char *str) {
 	return dest + len;
 }
 
-int connmgr_init(const char* filter) {
-	struct mime_type *entry;
+int connmgr_init(void) {
```

Comment:
The problem I see with not passing the filter here as initialization parameter is that we're now implicitly dependent on the initialization sequence; someone must have called `connmgr_set_mime_filter()` before for this to work properly.

The name `connmgr_init()` implies, that this is the first function to be called - but it only works correctly if the mime filter was set before. It is very easy to miss this implied need for this, and some refactoring could easily break it
If the parameter is given here, then it is impossible to use wrongly as the caller needs to know where they get the filter from. Like with any API, it is good to design it in a way that it is hard to mis-use.

So I'd like to see this still taking the filter parameter (and then internally call  a static function `set_mime_filter()`

Previously, we initialized by passing through the renderer, maybe that should be done (it moves the problem one up though - now we need to know to initialize the renderer with the filter). Maybe you can see a way to make this robust where the API implies its use ?
(Longer term, modeling this nicer in way that clearly indicates that the output (e.g. output_gstreamer) is the one providing the supported mime types in the first place is also a goal, but not now).

Reply:
Fair point. To get the filter into connmgr_init we could
* Change the function prototype for the device `init_function` to take an argument and then pass the MIME filter through `upnp_device_init`.

OR

* Add MIME filter as an argument to `upnp_renderer_descriptor` (which kinda acts like a constructor) to assign the MIME filter to the `upnp_device_descriptor` and pass the filter forward as previously implemented in master.

Reply:
The first option sounds interesting, but it also would mean that something that is meant to be generic (as `upnp_device_descriptor` implies) now take a `const char*` parameter, which through that indirection does not really indicate anymore what that parameter is. On the plus side, this is the only one.

The second one has the advantage that it has a properly named parameter, but the disadvantage is similar to my original complaint: now we need to magically first call the function to set the filter before everything else.

So both options have their issues, but I think I slightly prefer the first one.

(I plan to convert the code-base to C++ at some point, which makes it much easier to do things expressively).